packages/smooth_app/lib/generic_lib/loading_sliver.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:flutter/widgets.dart';
+
+class LoadingSliverChildBuilderDelegate extends SliverChildBuilderDelegate {
+  /// Represents a ListView that can show progress by displaying
+  /// [loadingWidget]s.
+  ///
+  /// When [loading] is `true`, [loadingCount] of [loadingWidget] will be
+  /// displayed.
```

Comment:
1. the comments should be on top of the class
2. I believe the comments are wrong: it's not a `ListView`, and it's about a `SliverList` anyway
3. Not convinced by the use of a class whose code is simple and is used only once (AFAIK)
4. the filename has little resemblance with the class name, and that is annoying

Reply:
1. Done!
2. Done!
3. I extracted the class to make sure that we can reuse it instead of wrapping each time a SliverChildBuilderDelegate with a "loading shimmer". If you feel that this could be the only case I'll inline it.
4. I took inspiration from the flutter library, where the file containing every class related to a Sliver in just called sliver.dart. Not a big requirement from me, I can change it.

Reply:
Thank you; please inline it.